h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -487,13 +504,15 @@ private int nextInstanceCounter(String algoName, NonBlockingHashMap<String, Inte
    * Helper for hex.ModelBuilder.
    * @return
    */
-  public Job trainModel(Key<Model> key, String algoURLName, Model.Parameters parms) {
-    String algoName = ModelBuilder.algoName(algoURLName);
+  public Job<Model> trainModel(Key<Model> key, Algo algo, Model.Parameters parms) {
+    if (exceededSearchLimits(algo, key == null ? null : key.toString(), JobType.ModelBuild)) return null;
```

Comment:
I'm considering throwing an exception instead of returning null to avoid useless computations, need to see the impact first.